packages/code-combinators/test/Test/CodeCombinators/GenPat.hs

```diff
@@ -0,0 +1,106 @@
+module Test.CodeCombinators.GenPat where
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Language.Haskell.TH as TH
+import qualified Happy.Backend.CodeCombinators.Syntax as SnGen
+import qualified Happy.Backend.CodeCombinators.Abstract()
+import Happy.Backend.CodeCombinators
+import Test.CodeCombinators.Common
+import Data.List
+
+
+genLitP :: MonadGen m => m TH.Pat
+genLitP = do
+  x <- Gen.choice $
+         [
+            do
+            c <- Gen.latin1
+            return $ litP $ TH.CharL c
+          , do
+            s <- Gen.list (Range.linear 0 20) Gen.latin1
+            return $ litP $ TH.StringL $ delete '\n' s
+          , do
+            x <- Gen.int $ Range.linear minBound maxBound
+            return $ litP $ TH.IntegerL $ fromIntegral x
+         ]
+  return x
+
+genBoolP :: MonadGen m => m TH.Pat
+genBoolP = Gen.element [trueP, falseP]
+
+genVarP :: MonadGen m => m TH.Pat
+genVarP = do
+  varName <- genFunName
+  return $ varP varName
```

Comment:
```suggestion
genVarP = varP <$> genFunName
```

Reply:
With the above change, I’m not even sure that you need this definition at all, you could just inline it in the generator where is is used.